### PR TITLE
Add devcontainer and Build Tool

### LINKED ISSUE #3: Introduce Dev Container
- Python 3.11
- Extensions
-

### PR DESCRIPTION
**Pull request by macromania**
Fixes #3 and #4

### CODE REVIEW

File: .devcontainer/devcontainer.json
Comment (macromania) [inline .devcontainer/devcontainer.json:1]: A simple start. If we need further customisation, we can add `Dockerfile` as we need.

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
+  "python.testing.pytestArgs": [
+    "test"
+  ],
   "python.testing.unittestEnabled": false,
-  "python.testing.pytestEnabled": true
-}
+  "python.testing.pytestEnabled": true,
+  "python.terminal.activateEnvironment": true,
+  "python.terminal.activateEnvInCurrentTerminal": true,
+  "python.defaultInterpreterPath": ".venv/bin/python",
```
Comment (macromania) [inline .vscode/settings.json:9]: Makes sure the environment created inside the devcontainer is set in the VSCode so the developer doesn't need to remember...

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+[tool.poetry.group.dev.dependencies]
+pytest = "^8.1.1"
+
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.ruff]
+line-length = 240
```
Comment (macromania) [inline pyproject.toml:20]: We all use wide monitors, and unless the otherwise is convinced, I think we go bigger 😏

File: requirements.txt
Comment (macromania) [inline requirements.txt:1]: Now this is in `pyproject.toml`

File: scripts/run-unit-tests.sh
Comment (macromania) [inline scripts/run-unit-tests.sh:1]: make them executable.

File: scripts/setup-local.sh
Comment (macromania) [inline scripts/setup-local.sh:1]: make them executable.

File: src/new_experiment.py
Comment (macromania) [inline src/new_experiment.py:1]: ruff and linting in place. No logic updates.

**Review by macromania — COMMENTED** [review]
@jedheaj314  ready for review :)

**Comment by jedheaj314** [comment]
LGTM.
nit/minor: should we add new line at the end of the files ( devcontainer.json, postCreateCommand.sh etc )
Also should we change Readme > Local Development instructions?

**Review by jedheaj314 — APPROVED** [review]

**Pull request merged**